### ISSUE TITLE
Invalid IBAN Considered Valid

### ISSUE THREAD

**Issue by umangkedia92** [arhs/iban.js#72]
According to https://www.iban.com/testibans the following IBANs are invalid:
GB00HLFX11016111455365
GB01BARC20714583608387

But this library considers these IBANs as valid.

**Comment by floSoX** [comment]
According to https://runkit.com/embed/y2e8cpx51ogy, the bad list is larger:
- GB00HLFX11016111455365
- GB01BARC20714583608387
- GB02BARC20201530093451
- GB24BARC20201630093459
- GB2LABBY09012857201707
- GB68CITI18500483515538

```js
var IBAN = require("iban")
const iban_list = ['GB33BUKB20201555555555', 'GB94BARC10201530093459', 'GB94BARC20201530093459', 'GB96BARC202015300934591', 'GB02BARC20201530093451', 'GB68CITI18500483515538', 'GB24BARC20201630093459', 'GB12BARC20201530093A59', 'GB78BARCO0201530093459', 'GB2LABBY09012857201707', 'GB01BARC20714583608387', 'GB00HLFX11016111455365', 'US64SVBKUS6S3300958879'];
let res = [];
iban_list.forEach(function(iban) {
  res[iban] = IBAN.isValid(iban);
});
res;
```

**Comment by LaurentVB** [comment]
Hello,
Thanks for opening this issue.
You're right that this library should probably have the same results as documented on iban.com for the following IBANs:
- GB00HLFX11016111455365
- GB01BARC20714583608387
- GB2LABBY09012857201707

The other examples either give the same answer as iban.com, or expectedly return `true` when iban.com replies `false` because of local validation criteria not checked by this library (bank code, bban check-digit).

**Comment by Renkas** [comment]
also lowercase country code is considered valid although standard specifies that all letters in IBAN MUST be uppercase.

**Comment by peter-catalin** [comment]
Hello,

I found another case that is being considered valid by the library but invalid on iban.com

ES1100753038790600500941

Hope it helps!

**Comment by puru0019** [comment]
Hi 

iban is allowing special characters and telling it as valid

![image](https://user-images.githubusercontent.com/23036485/81553336-8dba4000-93a2-11ea-9e54-8066343f5fbe.png)

**Comment by awacode21** [comment]
This is also a problem in my current customer project where we are using the Iban library. Can you already estimate when this will be fixed?

**Comment by syshex** [comment]
yeah we also got this problem. Accepting chars like  '?'  ,   '.'   etc in the middle of the iban.

**Comment by tarmooo** [comment]
> yeah we also got this problem. Accepting chars like '?' , '.' etc in the middle of the iban.

Why you even let user enter those chars, clean it up before sending it to validator? I think this library should not be responsible for cleanup.

**Comment by syshex** [comment]
> > yeah we also got this problem. Accepting chars like '?' , '.' etc in the middle of the iban.
> 
> Why you even let user enter those chars, clean it up before sending it to validator? I think this library should not be responsible for cleanup.

Hey, yes, we are validating inputs before. 

Still, I do not agree with your comment.  In the main page there is this: `IBAN.isValid('hello world'); // false`      , space is an invalid char , yet, the library works fine, by returning false.   A false positive on the other hand is definitely not a good return , even if input should have been sanitized before.  It is the same as saying   : `IBAN.isValid('hello world'); // true`   because the input should have been checked for the space.

**Comment by Cyrille37** [comment]
> You're right that this library should probably have the same results as documented on iban.com for the following IBANs:
>     * GB00HLFX11016111455365
>     * GB01BARC20714583608387
>     * GB2LABBY09012857201707

That's mean there is a error in ckecksum calculation ...

**Comment by KyorCode** [comment]
There actualy is a problem with the IBAN Checksum:

BE77776593869767
Is also valid, although not by iban.com.

**Comment by tarmooo** [comment]
same goes for ES8350210036679521296135 - fails for bank code checksum
iban.com says it is invalid. It seems this library does not count on spec per country.

**Comment by WebDaMa** [comment]
Perhaps a simple fix is that you run this on your input IBAN string before validating:
```javascript
input = input.replace(/[^a-zA-Z0-9]/g, '');
```
This way all weird characters are filtered out, perhaps it could be included in this script as well?
